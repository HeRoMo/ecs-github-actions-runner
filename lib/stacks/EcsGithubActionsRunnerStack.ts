import { Stack } from 'aws-cdk-lib';
import { Vpc } from 'aws-cdk-lib/aws-ec2';
import { Construct } from 'constructs';
import { EcsGithubActionsRunner } from '../constructs/EcsGithubActionsRunner';

/**
 * Construct ECS cluster of Github actions runner.
 */
export class EcsGithubActionsRunnerStack extends Stack {
  private readonly clusterConstructor: EcsGithubActionsRunner;

  constructor(scope: Construct, clusterName: string, private readonly vpc: Vpc) {
    super(scope, `${clusterName}-stack`);

    this.clusterConstructor = new EcsGithubActionsRunner(this, clusterName, vpc);
  }

  /**
   * runnerTaskInfo
   */
  public get runnerTaskInfo() {
    return this.clusterConstructor.clusterInfo;
  }
}
