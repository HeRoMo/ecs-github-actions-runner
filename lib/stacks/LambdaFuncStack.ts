import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';

import { CONFIG } from '../Config';
import { EcsGithubActionsRunnerInfo } from '../constructs/EcsGithubActionsRunner';
import { LambdaConstructor } from '../constructs/LambdaConstructor';

const props: StackProps = {
  env: { region: CONFIG.region },
  description: `Lambda functions to operate Tasks of ${CONFIG.clusterName}`,
};

/**
 * Construnt Lambda functions to operate Github runner tasks.
 */
export class LambdaFuncStack extends Stack {
  constructor(scope: Construct, id: string, taskInfo: EcsGithubActionsRunnerInfo) {
    super(scope, `${id}-stack`, props);

    new LambdaConstructor(this, id, taskInfo);
  }
}
