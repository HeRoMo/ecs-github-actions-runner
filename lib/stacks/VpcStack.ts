import { Stack, StackProps } from 'aws-cdk-lib';
import {
  SubnetConfiguration, SubnetType, Vpc, VpcProps,
} from 'aws-cdk-lib/aws-ec2';
import { Construct } from 'constructs';

/**
 * Construct VPC stack
 */
export class VpcStack extends Stack {
  public readonly vpc: Vpc;
  constructor(scope: Construct, vpcName: string, props?: StackProps | undefined) {
    super(scope, `${vpcName}-stack`, props)

    const subnetConfiguration: SubnetConfiguration[] = [
      { name: 'Public', cidrMask: 24, subnetType: SubnetType.PUBLIC },
    ];

    const vpcProps: VpcProps = {
      vpcName,
      cidr: '10.0.0.0/16',
      vpnGateway: false,
      natGateways: 0,
      subnetConfiguration,
    };
    this.vpc = new Vpc(this, vpcName, vpcProps);
  }
}
