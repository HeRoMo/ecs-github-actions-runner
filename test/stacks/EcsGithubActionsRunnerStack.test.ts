import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';

import { EcsGithubActionsRunnerStack } from '../../lib/stacks/EcsGithubActionsRunnerStack';
import { VpcStack } from '../../lib/stacks/VpcStack';

test('Snapshot Test', () => {
  const app = new cdk.App();
  const vpcStack = new VpcStack(app, 'test-vpc');
  const runnerStack = new EcsGithubActionsRunnerStack(app, 'test-runner', vpcStack.vpc);

  expect(Template.fromStack(runnerStack)).toMatchSnapshot();
});
