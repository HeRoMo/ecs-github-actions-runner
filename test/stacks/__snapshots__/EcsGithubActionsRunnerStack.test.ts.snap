// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 1`] = `
Object {
  "Description": "github-actions-runner, ECS cluster for Github self-hosted runner",
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValueawsserviceecsoptimizedamiamazonlinux2arm64recommendedimageidC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "/aws/service/ecs/optimized-ami/amazon-linux-2/arm64/recommended/image_id",
      "Type": "AWS::SSM::Parameter::Value<AWS::EC2::Image::Id>",
    },
    "SsmParameterValueawsserviceecsoptimizedamiamazonlinux2recommendedimageidC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "/aws/service/ecs/optimized-ami/amazon-linux-2/recommended/image_id",
      "Type": "AWS::SSM::Parameter::Value<AWS::EC2::Image::Id>",
    },
  },
  "Resources": Object {
    "testrunner3196B9C4": Object {
      "Properties": Object {
        "CapacityProviders": Array [
          "FARGATE",
          "FARGATE_SPOT",
          Object {
            "Ref": "testrunnerAsgCapacityProvideramd647BF90295",
          },
          Object {
            "Ref": "testrunnerAsgCapacityProviderarm6469ABD66E",
          },
        ],
        "Cluster": Object {
          "Ref": "testrunner60D7B13B",
        },
        "DefaultCapacityProviderStrategy": Array [],
      },
      "Type": "AWS::ECS::ClusterCapacityProviderAssociations",
    },
    "testrunner60D7B13B": Object {
      "Properties": Object {
        "ClusterName": "test-runner",
      },
      "Type": "AWS::ECS::Cluster",
    },
    "testrunnerASGamd64ASG477C7D76": Object {
      "Properties": Object {
        "AutoScalingGroupName": "test-runner-amd64",
        "LaunchConfigurationName": Object {
          "Ref": "testrunnerASGamd64LaunchConfigF11DA921",
        },
        "MaxSize": "1",
        "MinSize": "0",
        "NewInstancesProtectedFromScaleIn": true,
        "Tags": Array [
          Object {
            "Key": "Name",
            "PropagateAtLaunch": true,
            "Value": "test-runner-stack/test-runner/ASG-amd64",
          },
        ],
        "TerminationPolicies": Array [
          "OldestInstance",
          "Default",
        ],
        "VPCZoneIdentifier": Array [
          Object {
            "Fn::ImportValue": "test-vpc-stack:ExportsOutputReftestvpcPublicSubnet1Subnet01CF7554F3858355",
          },
          Object {
            "Fn::ImportValue": "test-vpc-stack:ExportsOutputReftestvpcPublicSubnet2Subnet4E9D9728B25B3FFC",
          },
        ],
      },
      "Type": "AWS::AutoScaling::AutoScalingGroup",
      "UpdatePolicy": Object {
        "AutoScalingScheduledAction": Object {
          "IgnoreUnmodifiedGroupSizeProperties": true,
        },
      },
    },
    "testrunnerASGamd64InstanceProfile828F9683": Object {
      "Properties": Object {
        "Roles": Array [
          Object {
            "Ref": "testrunnerASGamd64InstanceRole5BE4E643",
          },
        ],
      },
      "Type": "AWS::IAM::InstanceProfile",
    },
    "testrunnerASGamd64InstanceRole5BE4E643": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ec2.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "test-runner-stack/test-runner/ASG-amd64",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "testrunnerASGamd64InstanceRoleDefaultPolicy878D6338": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "ecs:DeregisterContainerInstance",
                "ecs:RegisterContainerInstance",
                "ecs:Submit*",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "testrunner60D7B13B",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": Array [
                "ecs:Poll",
                "ecs:StartTelemetrySession",
              ],
              "Condition": Object {
                "ArnEquals": Object {
                  "ecs:cluster": Object {
                    "Fn::GetAtt": Array [
                      "testrunner60D7B13B",
                      "Arn",
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "ecs:DiscoverPollEndpoint",
                "ecr:GetAuthorizationToken",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "testrunnerASGamd64InstanceRoleDefaultPolicy878D6338",
        "Roles": Array [
          Object {
            "Ref": "testrunnerASGamd64InstanceRole5BE4E643",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "testrunnerASGamd64LaunchConfigF11DA921": Object {
      "DependsOn": Array [
        "testrunnerASGamd64InstanceRoleDefaultPolicy878D6338",
        "testrunnerASGamd64InstanceRole5BE4E643",
      ],
      "Properties": Object {
        "IamInstanceProfile": Object {
          "Ref": "testrunnerASGamd64InstanceProfile828F9683",
        },
        "ImageId": Object {
          "Ref": "SsmParameterValueawsserviceecsoptimizedamiamazonlinux2recommendedimageidC96584B6F00A464EAD1953AFF4B05118Parameter",
        },
        "InstanceType": "c6i.large",
        "KeyName": "your-key-name",
        "SecurityGroups": Array [
          Object {
            "Fn::GetAtt": Array [
              "testrunnertestrunnersgFD82329D",
              "GroupId",
            ],
          },
        ],
        "SpotPrice": "0.107",
        "UserData": Object {
          "Fn::Base64": Object {
            "Fn::Join": Array [
              "",
              Array [
                "#!/bin/bash
echo ECS_CLUSTER=",
                Object {
                  "Ref": "testrunner60D7B13B",
                },
                " >> /etc/ecs/ecs.config
sudo iptables --insert FORWARD 1 --in-interface docker+ --destination 169.254.169.254/32 --jump DROP
sudo service iptables save
echo ECS_AWSVPC_BLOCK_IMDS=true >> /etc/ecs/ecs.config",
              ],
            ],
          },
        },
      },
      "Type": "AWS::AutoScaling::LaunchConfiguration",
    },
    "testrunnerASGarm64ASG1AC09258": Object {
      "Properties": Object {
        "AutoScalingGroupName": "test-runner-arm64",
        "LaunchConfigurationName": Object {
          "Ref": "testrunnerASGarm64LaunchConfigB2F470A4",
        },
        "MaxSize": "1",
        "MinSize": "0",
        "NewInstancesProtectedFromScaleIn": true,
        "Tags": Array [
          Object {
            "Key": "Name",
            "PropagateAtLaunch": true,
            "Value": "test-runner-stack/test-runner/ASG-arm64",
          },
        ],
        "TerminationPolicies": Array [
          "OldestInstance",
          "Default",
        ],
        "VPCZoneIdentifier": Array [
          Object {
            "Fn::ImportValue": "test-vpc-stack:ExportsOutputReftestvpcPublicSubnet1Subnet01CF7554F3858355",
          },
          Object {
            "Fn::ImportValue": "test-vpc-stack:ExportsOutputReftestvpcPublicSubnet2Subnet4E9D9728B25B3FFC",
          },
        ],
      },
      "Type": "AWS::AutoScaling::AutoScalingGroup",
      "UpdatePolicy": Object {
        "AutoScalingScheduledAction": Object {
          "IgnoreUnmodifiedGroupSizeProperties": true,
        },
      },
    },
    "testrunnerASGarm64InstanceProfile8AD29D16": Object {
      "Properties": Object {
        "Roles": Array [
          Object {
            "Ref": "testrunnerASGarm64InstanceRole4B7031AE",
          },
        ],
      },
      "Type": "AWS::IAM::InstanceProfile",
    },
    "testrunnerASGarm64InstanceRole4B7031AE": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ec2.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "test-runner-stack/test-runner/ASG-arm64",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "testrunnerASGarm64InstanceRoleDefaultPolicy01D9960F": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "ecs:DeregisterContainerInstance",
                "ecs:RegisterContainerInstance",
                "ecs:Submit*",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "testrunner60D7B13B",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": Array [
                "ecs:Poll",
                "ecs:StartTelemetrySession",
              ],
              "Condition": Object {
                "ArnEquals": Object {
                  "ecs:cluster": Object {
                    "Fn::GetAtt": Array [
                      "testrunner60D7B13B",
                      "Arn",
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "ecs:DiscoverPollEndpoint",
                "ecr:GetAuthorizationToken",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "testrunnerASGarm64InstanceRoleDefaultPolicy01D9960F",
        "Roles": Array [
          Object {
            "Ref": "testrunnerASGarm64InstanceRole4B7031AE",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "testrunnerASGarm64LaunchConfigB2F470A4": Object {
      "DependsOn": Array [
        "testrunnerASGarm64InstanceRoleDefaultPolicy01D9960F",
        "testrunnerASGarm64InstanceRole4B7031AE",
      ],
      "Properties": Object {
        "IamInstanceProfile": Object {
          "Ref": "testrunnerASGarm64InstanceProfile8AD29D16",
        },
        "ImageId": Object {
          "Ref": "SsmParameterValueawsserviceecsoptimizedamiamazonlinux2arm64recommendedimageidC96584B6F00A464EAD1953AFF4B05118Parameter",
        },
        "InstanceType": "c6g.large",
        "KeyName": "your-key-name",
        "SecurityGroups": Array [
          Object {
            "Fn::GetAtt": Array [
              "testrunnertestrunnersgFD82329D",
              "GroupId",
            ],
          },
        ],
        "SpotPrice": "0.0856",
        "UserData": Object {
          "Fn::Base64": Object {
            "Fn::Join": Array [
              "",
              Array [
                "#!/bin/bash
echo ECS_CLUSTER=",
                Object {
                  "Ref": "testrunner60D7B13B",
                },
                " >> /etc/ecs/ecs.config
sudo iptables --insert FORWARD 1 --in-interface docker+ --destination 169.254.169.254/32 --jump DROP
sudo service iptables save
echo ECS_AWSVPC_BLOCK_IMDS=true >> /etc/ecs/ecs.config",
              ],
            ],
          },
        },
      },
      "Type": "AWS::AutoScaling::LaunchConfiguration",
    },
    "testrunnerAsgCapacityProvideramd647BF90295": Object {
      "Properties": Object {
        "AutoScalingGroupProvider": Object {
          "AutoScalingGroupArn": Object {
            "Ref": "testrunnerASGamd64ASG477C7D76",
          },
          "ManagedScaling": Object {
            "Status": "ENABLED",
            "TargetCapacity": 100,
          },
          "ManagedTerminationProtection": "ENABLED",
        },
        "Name": "amd64-test-runner-asg-cp",
      },
      "Type": "AWS::ECS::CapacityProvider",
    },
    "testrunnerAsgCapacityProviderarm6469ABD66E": Object {
      "Properties": Object {
        "AutoScalingGroupProvider": Object {
          "AutoScalingGroupArn": Object {
            "Ref": "testrunnerASGarm64ASG1AC09258",
          },
          "ManagedScaling": Object {
            "Status": "ENABLED",
            "TargetCapacity": 100,
          },
          "ManagedTerminationProtection": "ENABLED",
        },
        "Name": "arm64-test-runner-asg-cp",
      },
      "Type": "AWS::ECS::CapacityProvider",
    },
    "testrunnertestrunnerlgBD0B88CA": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "LogGroupName": "/ecs/test-runner",
        "RetentionInDays": 7,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Retain",
    },
    "testrunnertestrunnersgFD82329D": Object {
      "Properties": Object {
        "GroupDescription": "SG for test-runner",
        "GroupName": "test-runner-sg",
        "SecurityGroupEgress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound traffic by default",
            "IpProtocol": "-1",
          },
        ],
        "SecurityGroupIngress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow SSH Access",
            "FromPort": 22,
            "IpProtocol": "tcp",
            "ToPort": 22,
          },
        ],
        "VpcId": Object {
          "Fn::ImportValue": "test-vpc-stack:ExportsOutputReftestvpc8985080E5120E245",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "testrunnertestrunnertdec279C141CD": Object {
      "Properties": Object {
        "ContainerDefinitions": Array [
          Object {
            "Environment": Array [
              Object {
                "Name": "REPOSITORY_URL",
                "Value": "https://github.com/your/reponame",
              },
              Object {
                "Name": "TOKEN",
                "Value": "set_self-hosted_runner_token",
              },
            ],
            "Essential": true,
            "Image": "ghcr.io/heromo/ecs-github-actions-runner:latest",
            "LogConfiguration": Object {
              "LogDriver": "awslogs",
              "Options": Object {
                "awslogs-group": Object {
                  "Ref": "testrunnertestrunnerlgBD0B88CA",
                },
                "awslogs-region": "us-east-1",
                "awslogs-stream-prefix": "github-actions-runner",
              },
            },
            "Memory": 3500,
            "MountPoints": Array [
              Object {
                "ContainerPath": "/var/run/docker.sock",
                "ReadOnly": true,
                "SourceVolume": "docker_sock",
              },
            ],
            "Name": "github-actions-runner",
          },
        ],
        "ExecutionRoleArn": Object {
          "Fn::GetAtt": Array [
            "testrunnertestrunnertdec2ExecutionRoleE89C3F52",
            "Arn",
          ],
        },
        "Family": "test-runner-ec2",
        "NetworkMode": "bridge",
        "RequiresCompatibilities": Array [
          "EC2",
        ],
        "TaskRoleArn": Object {
          "Fn::GetAtt": Array [
            "testrunnertestrunnertdec2TaskRole080BB98E",
            "Arn",
          ],
        },
        "Volumes": Array [
          Object {
            "Host": Object {
              "SourcePath": "/var/run/docker.sock",
            },
            "Name": "docker_sock",
          },
        ],
      },
      "Type": "AWS::ECS::TaskDefinition",
    },
    "testrunnertestrunnertdec2ExecutionRoleDefaultPolicy52E588AD": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "testrunnertestrunnerlgBD0B88CA",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "testrunnertestrunnertdec2ExecutionRoleDefaultPolicy52E588AD",
        "Roles": Array [
          Object {
            "Ref": "testrunnertestrunnertdec2ExecutionRoleE89C3F52",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "testrunnertestrunnertdec2ExecutionRoleE89C3F52": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "testrunnertestrunnertdec2TaskRole080BB98E": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "testrunnertestrunnertdfgarm648BFB5B4E": Object {
      "Properties": Object {
        "ContainerDefinitions": Array [
          Object {
            "Environment": Array [
              Object {
                "Name": "REPOSITORY_URL",
                "Value": "https://github.com/your/reponame",
              },
              Object {
                "Name": "TOKEN",
                "Value": "set_self-hosted_runner_token",
              },
            ],
            "Essential": true,
            "Image": "ghcr.io/heromo/ecs-github-actions-runner:latest",
            "LogConfiguration": Object {
              "LogDriver": "awslogs",
              "Options": Object {
                "awslogs-group": Object {
                  "Ref": "testrunnertestrunnerlgBD0B88CA",
                },
                "awslogs-region": "us-east-1",
                "awslogs-stream-prefix": "github-actions-runner",
              },
            },
            "Memory": 3500,
            "Name": "github-actions-runner",
          },
        ],
        "Cpu": "512",
        "ExecutionRoleArn": Object {
          "Fn::GetAtt": Array [
            "testrunnertestrunnertdfgarm64ExecutionRoleCD2346EA",
            "Arn",
          ],
        },
        "Family": "test-runner-fg-arm64",
        "Memory": "1024",
        "NetworkMode": "awsvpc",
        "RequiresCompatibilities": Array [
          "FARGATE",
        ],
        "RuntimePlatform": Object {
          "CpuArchitecture": "ARM64",
          "OperatingSystemFamily": "LINUX",
        },
        "TaskRoleArn": Object {
          "Fn::GetAtt": Array [
            "testrunnertestrunnertdfgarm64TaskRoleABE70156",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ECS::TaskDefinition",
    },
    "testrunnertestrunnertdfgarm64ExecutionRoleCD2346EA": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "testrunnertestrunnertdfgarm64ExecutionRoleDefaultPolicy5FA2A402": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "testrunnertestrunnerlgBD0B88CA",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "testrunnertestrunnertdfgarm64ExecutionRoleDefaultPolicy5FA2A402",
        "Roles": Array [
          Object {
            "Ref": "testrunnertestrunnertdfgarm64ExecutionRoleCD2346EA",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "testrunnertestrunnertdfgarm64TaskRoleABE70156": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "testrunnertestrunnertdfgx86647A3571A7": Object {
      "Properties": Object {
        "ContainerDefinitions": Array [
          Object {
            "Environment": Array [
              Object {
                "Name": "REPOSITORY_URL",
                "Value": "https://github.com/your/reponame",
              },
              Object {
                "Name": "TOKEN",
                "Value": "set_self-hosted_runner_token",
              },
            ],
            "Essential": true,
            "Image": "ghcr.io/heromo/ecs-github-actions-runner:latest",
            "LogConfiguration": Object {
              "LogDriver": "awslogs",
              "Options": Object {
                "awslogs-group": Object {
                  "Ref": "testrunnertestrunnerlgBD0B88CA",
                },
                "awslogs-region": "us-east-1",
                "awslogs-stream-prefix": "github-actions-runner",
              },
            },
            "Memory": 3500,
            "Name": "github-actions-runner",
          },
        ],
        "Cpu": "512",
        "ExecutionRoleArn": Object {
          "Fn::GetAtt": Array [
            "testrunnertestrunnertdfgx8664ExecutionRoleC2D3A897",
            "Arn",
          ],
        },
        "Family": "test-runner-fg-x86_64",
        "Memory": "1024",
        "NetworkMode": "awsvpc",
        "RequiresCompatibilities": Array [
          "FARGATE",
        ],
        "RuntimePlatform": Object {
          "CpuArchitecture": "X86_64",
          "OperatingSystemFamily": "LINUX",
        },
        "TaskRoleArn": Object {
          "Fn::GetAtt": Array [
            "testrunnertestrunnertdfgx8664TaskRoleAC6A46D8",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ECS::TaskDefinition",
    },
    "testrunnertestrunnertdfgx8664ExecutionRoleC2D3A897": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "testrunnertestrunnertdfgx8664ExecutionRoleDefaultPolicy4754446E": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "testrunnertestrunnerlgBD0B88CA",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "testrunnertestrunnertdfgx8664ExecutionRoleDefaultPolicy4754446E",
        "Roles": Array [
          Object {
            "Ref": "testrunnertestrunnertdfgx8664ExecutionRoleC2D3A897",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "testrunnertestrunnertdfgx8664TaskRoleAC6A46D8": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
